### ISSUE TITLE
Access bigwigmerge from other crates

### ISSUE THREAD

**Issue by donaldcampbelljr** [jackh726/bigtools#59]
Hi @jackh726,

Thank you for your work on bigtools!

I am currently exploring using bigtools to create bigwig files in one of our rust crates (https://github.com/databio/gtars). Our crate would create intermediate bigwigs before merging them into a single bigwig as a final result.

I thought the best way would be to call the the public function `bigwigmerge` from `bigwigmerge.rs` and pass a constructed `BigWigMergeArgs` struct to the function. However, while the `BigWigMergeArgs` struct is public, some of the fields are private and I am unable to edit these fields from within another crate.

I also looked at a second option, where I merge the values and write them using `BigWigWrite<File>`'s wrtie function:
```
            let all_values = ChromGroupReadImpl {
                iter: Box::new(iter),
            };
            outb.write(all_values, runtime)?;
```

However, all values is a _private_ struct that implements the necessary `BBIDataSource` trait.

I guess my questions for you are:

1. Should I be approaching this differently?  I could have the user run `bigwigmerge [OPTIONS] <OUTPUT>` after using our crate, but I thought it would be nicer and less clunky to do this behind the scenes for the user.
2. Could we make the ChromGroupReadImpl struct public? I believe this might be the simplest solution to using the bigtools merge code  in another external crate.

**Comment by jackh726** [comment]
Hi @donaldcampbelljr!

The right thing to do is for me to just make the available functions and structs public. Happy to take a PR for this - otherwise I will try to get to it maybe this evening or something this week.

**Comment by donaldcampbelljr** [comment]
Great, I've created a PR with the necessary changes. I've tested my forked branch in our external crate and have confirmed that they fix the problem I was having.

https://github.com/jackh726/bigtools/pull/60

**Comment by jackh726** [comment]
Merged. Will keep this open until the next release. (Will make a release cut soonish.)

**Comment by donaldcampbelljr** [comment]
Awesome. Thank you!

**Comment by jackh726** [comment]
This is included in bigtools 0.5.4 released today!

Closing - let me know if you run into other issues!
